api/src/middlewares/services/StaffService/StaffService.ts

```diff
@@ -169,7 +170,14 @@ export class StaffService {
     const emailAddress = getPostEmailAddress(country, post);
     const personalisation = PersonalisationBuilder.postNotification(answers, type);
     if (!emailAddress) {
-      this.logger.warn(`No email address found for country ${country} - post ${post}. Post notification will not be sent`);
+      this.logger.error(
+        new ApplicationError(
```

Comment:
Let's just log the error but use the code so it's easily filterable. ApplicationError is mainly used in the API so that it can be caught by the express error handlers and set the error code based on what was thrown.